db/rdbms/src/main/resources/mapper/VariableMapper.xml

```diff
@@ -9,30 +9,123 @@
 <!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
 <mapper namespace="io.camunda.db.rdbms.sql.VariableMapper">
 
-  <select id="findOne"
-    statementType="PREPARED"
-    parameterType="java.lang.Long"
-    resultType="io.camunda.db.rdbms.write.domain.VariableDbModel">
-      SELECT
-        VAR_KEY,
-        PROCESS_INSTANCE_KEY,
-        SCOPE_KEY,
-        VAR_NAME,
-        VAR_VALUE,
-        TRUNCATED,
-        TENANT_ID
-      FROM
-        VARIABLE
-      WHERE VAR_KEY = #{key}
+  <select id="count" resultType="java.lang.Long">
+    SELECT COUNT(*)
+    FROM VARIABLE
+    <include refid="io.camunda.db.rdbms.sql.VariableMapper.searchFilter"/>
   </select>
 
+  <!-- default search statement for databases supporting LIMIT/OFFSET-->
+  <select id="search" parameterType="io.camunda.db.rdbms.read.domain.VariableDbQuery" resultMap="io.camunda.db.rdbms.sql.VariableMapper.searchResultMap">
+    SELECT
+      VAR_KEY,
+      PROCESS_INSTANCE_KEY,
+      SCOPE_KEY,
+      VAR_NAME,
+      VAR_VALUE,
+      TENANT_ID,
+      false as IS_PREVIEW
```

Comment:
I see this isPreview in the entity but not in any filter.

Do we need a follow-up ticket for clarification/implementation?

Reply:
Yea I think we need to discuss this. Since we do NOT save truncated date, it's never a preview ... I'm not sure if the webapps need a preview ... I'll clarify this.

Reply:
See here: https://github.com/camunda/camunda/issues/24230